terps/unp64/scanners/final_super_comp.cpp

```diff
@@ -0,0 +1,284 @@
+			unp->_idFlag = 1;
+			return;
+		}
+	}
+	/* SC/EqualSequences, FinalPack4/UA and TLC packer (sys2061/2064/2065/2066) */
+	if (unp->_depAdr == 0) {
+		for (p = 0x80d; p < 0x830; p++) {
+			if (mem[p + 2] == 0xb9) {
+				if (mem[p + 5] == 0x99 && mem[p + 0x15] == 0x4c &&
+					mem[p + 0x10] == mem[p + 0x81] &&
+					(*(unsigned int *)(mem + p + 0x78) == 0x18F7D0FC) &&
+					(*(unsigned int *)(mem + p + 0x8c) == 0xD0FCC4C8)) {
+					unp->_depAdr = READ_LE_UINT16(&mem[p + 0x16]);
+					for (q = 0x80b; q < 0x813; q++) {
+						if (mem[q] == 0x78) {
+							unp->_forced = q;
+							break;
+						}
+					}
+					if (q == 0x813) {
+						if (mem[p] == 0xa0)
+							unp->_forced = p;
+					}
+					unp->_endAdr = mem[p + 0x10];                  /* either 0x2d or 0xae */
+					unp->_strMem = READ_LE_UINT16(&mem[p + 0x2c]); /* $0800 fixed? */
+					p += 0x40;
+					while ((mem[p] != 0x4c) && (p < 0x860))
+						p++;
+					unp->_retAdr = READ_LE_UINT16(&mem[p + 1]);
+					break;
+				}
+			}
+		}
+		if (unp->_depAdr) {
+			unp->_idFlag = 1;
+			return;
+		}
+	}
+	/* SC/Equal chars */
+	if (unp->_depAdr == 0) {
+		if (((*(unsigned int *)(mem + 0x810) & 0xf0ffffff) == 0xA0A93878) &&
+			((*(unsigned int *)(mem + 0x814) == 0xFC852DE5) ||
+			 (*(unsigned int *)(mem + 0x814) == 0xFC85aeE5)) &&
+			((*(unsigned int *)(mem + 0x818) == 0xafE508A9) ||
+			 (*(unsigned int *)(mem + 0x818) == 0x2EE508A9)) &&
+			((*(unsigned int *)(mem + 0x844) & 0x00ffffff) == 0x00004C9A)) {
+			unp->_depAdr = READ_LE_UINT16(&mem[0x846]);
+			unp->_forced = 0x810;
+			unp->_retAdr = READ_LE_UINT16(&mem[0x872]);
+			unp->_endAdr = mem[0x866]; /* either 0x2d or 0xae */
+			unp->_strMem = READ_LE_UINT16(&mem[0x857]);
+			unp->_strMem += mem[0x849]; /* ldx #$?? -> sta $????,x; not always 0! */
+			unp->_idFlag = 1;
+			return;
+		}
+	}
+	/* FinalSuperCompressor/flexible hacked? */
+	if (unp->_depAdr == 0) {
+		if (((*(unsigned int *)(mem + 0x80d) & 0xff00ffff) == 0x9A00A278) &&
+			(*(unsigned int *)(mem + 0x814) == 0x9D0847BD) &&
+			(*(unsigned int *)(mem + 0x818) == 0x21BD0334) &&
+			(*(unsigned int *)(mem + 0x81c) == 0x040E9D09) &&
+			mem[0x878] == 0x4c) {
+			mem[0x80f] = 0xff; /* fixed, can't be otherwise */
+			unp->_depAdr = 0x334;
+			if (unp->_info->_run == -1)
+				unp->_forced = 0x80d;
+			unp->_retAdr = READ_LE_UINT16(&mem[0x879]);
+			/* patch */
+			mem[0x2a7] = 0x85;
+			mem[0x2a8] = 0x01;
+			mem[0x2a9] = 0x4c;
+			mem[0x2aa] = mem[0x879];
+			mem[0x2ab] = mem[0x87a];
+			unp->_strMem = 0x2a7;
+			unp->_fEndAf = 0x34e;
+			unp->_endAdC = 0xffff;
+			unp->_recurs = RECUMAX - 2;
+			unp->_idFlag = 1;
+			return;
+		}
+	}
+	/* FinalSuperCompressor/flexible hacked? 2nd layer */
+	if (unp->_depAdr == 0) {
+		if (unp->_info->_start == 0x2a7) {
+			if (mem[0x2a7] == 0x85 &&
+				mem[0x2a8] == 0x01 &&
+				mem[0x2a9] == 0x4c) {
+				p = READ_LE_UINT16(&mem[0x2aa]);
+				if ((*(unsigned int *)(mem + p) == ((((mem[0x2aa] + 0x15) & 0xff) << 24) | 0x00B99EA0)) &&
```

Comment:
Another signed/unsigned comparison warning in g++.  This can be addressed by making this constant unsigned, i.e. `0x00B99EA0U`.